dto/history.go

```diff
@@ -11,32 +11,36 @@ type HistoryRequest struct {
+	Data []*HistoryData `json:"data"`
 }
 
 type HistoryData struct {
-	LocationUUID        string          `json:"location_uuid"`
-	LocationName        string          `json:"location_name"`
-	LocationDescription string          `json:"location_description"`
-	GroupUUID           string          `json:"group_uuid"`
-	GroupName           string          `json:"group_name"`
-	GroupDescription    string          `json:"group_description"`
-	HostUUID            string          `json:"host_uuid"`
-	HostName            string          `json:"host_name"`
-	HostDescription     string          `json:"host_description"`
-	GlobalUUID          string          `json:"global_uuid"`
-	NetworkUUID         string          `json:"network_uuid"`
-	NetworkName         string          `json:"network_name"`
-	NetworkDescription  string          `json:"network_description"`
-	DeviceUUID          string          `json:"device_uuid"`
-	DeviceName          string          `json:"device_name"`
-	DeviceDescription   string          `json:"device_description"`
-	PointUUID           string          `json:"point_uuid"`
-	PointName           string          `json:"point_name"`
-	PointDescription    string          `json:"point_description"`
-	Values              []*HistoryValue `json:"values"`
+	LocationUUID        string            `json:"location_uuid"`
+	LocationName        string            `json:"location_name"`
+	LocationDescription string            `json:"location_description"`
+	GroupUUID           string            `json:"group_uuid"`
+	GroupName           string            `json:"group_name"`
+	GroupDescription    string            `json:"group_description"`
+	HostUUID            string            `json:"host_uuid"`
+	HostName            string            `json:"host_name"`
+	HostDescription     string            `json:"host_description"`
+	GlobalUUID          string            `json:"global_uuid"`
+	NetworkUUID         string            `json:"network_uuid"`
+	NetworkName         string            `json:"network_name"`
+	NetworkDescription  string            `json:"network_description"`
+	DeviceUUID          string            `json:"device_uuid"`
+	DeviceName          string            `json:"device_name"`
+	DeviceDescription   string            `json:"device_description"`
+	PointUUID           string            `json:"point_uuid"`
+	PointName           string            `json:"point_name"`
+	PointDescription    string            `json:"point_description"`
+	Values              []*HistoryValue   `json:"values"`
+	NetworkTags         []*HistoryTag     `json:"network_tags,omitempty"`
+	DeviceTags          []*HistoryTag     `json:"device_tags,omitempty"`
+	PointTags           []*HistoryTag     `json:"point_tags,omitempty"`
+	NetworkMetaTags     []*HistoryMetaTag `json:"network_meta_tags,omitempty"`
+	DeviceMetaTags      []*HistoryMetaTag `json:"device_meta_tags,omitempty"`
+	PointMetaTags       []*HistoryMetaTag `json:"point_meta_tags,omitempty"`
```

Comment:
As in this example: https://github.com/NubeIO/rubix-os/issues/791#issuecomment-2060429444

I guess it's better to have these fields in a single `tags` and `meta_tags`. We put:
- all the unique tags from network/device/point in the tags field and
- all unique combinations of keys and values from network/device/point in `meta_tags`.

Coz, we don't care where it's from, that's our convention -- it needs to search in the parent tree.

Otherwise, we want similar output in network/device/point's tags/meta_tags as well. And for that, we need to define a different `dto model` for that. For example:


```go
type Point struct {
        ... // all existing fields of point
	PointMetaTags   []*MetaTag `json:"point_meta_tags"`
	NetworkMetaTags []*MetaTag `json:"network_meta_tags"`
	DeviceMetaTags  []*MetaTag `json:"device_meta_tags"`
}
```